src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4766,15 +4759,15 @@ internal void EnsureCapacity(int capacity)
                 {
                     if (slot < Capacity)
                     {
-                        return _state[slot];
+                        return _state![slot];
```

Comment:
>_state! [](start = 31, length = 7)

Consider encapsulating `_state` through a private property to hide the fact that it could be null if not initialized.

Reply:
Wrapping this in a property seems like a bit of overhead for such a small type. And 3 of the 5 methods that use the field have to handle a null value.

---
In reply to: [241929195](https://github.com/dotnet/roslyn/pull/31719#discussion_r241929195) [](ancestors = 241929195)